### PR TITLE
BUG: Allow float64('1e10000') to overflow

### PR DESCRIPTION
**Pull request by eric-wieser**
Previously, this gave an error, but now it returns infinity like for the other types.
Unlike the other types, this emits a warning - it might be nice to do it for the other types too in future, or perhaps respect `np.seterr`

Fixes gh-9924.

Also some test cleanup while I was looking for where to put these - I'd suggesting looking commit-by-commit to make this clearer.

### CODE REVIEW

File: numpy/core/tests/test_scalar_ctors.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Test the scalar contructors, which also do type-coercion
+"""
+from __future__ import division, absolute_import, print_function
+
+import sys
+import numpy as np
+
+from numpy.testing import (
+    run_module_suite,
+    assert_equal, assert_almost_equal, assert_raises, assert_warns
```
Comment (mhvk) [inline numpy/core/tests/test_scalar_ctors.py]: Missing `,` at the end of this line

**Review by mhvk — COMMENTED** [review]
Looks good modulo the typo that causes all tests to fail.

**Comment by ahaldane** [comment]
Yep, looks right. 

One day we probably want to write our own `strtodl` too, since I've read it is pretty buggy in windows, but that is a lot of work.

**Comment by eric-wieser** [comment]
Looks like I accidentally flattened my commits together here too. Oops.

**Comment by eric-wieser** [comment]
Finally got it right, should be good to merge now

**Comment by ahaldane** [comment]
Great, LGTM.

Merging, thanks Eric!

**Comment by ahaldane** [comment]
We might want to rethink the warning here. 

First of all, we didn't think about underflow. `errno` gets set even for valid subnormal values: Eg (on x64):
```python
>>> np.float128('1.123e-4945')
RuntimeWarning: overflow encountered in conversion from string
1.123e-4945
```
The error message isn't quite right (should be underflow, not overflow), and it doesn't seem worth warning since it "works as expected", in some sense.

Even in the case of overflow to `inf`, the warning is inconsistent with the other float types as we discussed above. I think we should make things consistent: Either warn on overflow for all float types, or do it for none. 

For 1.14.1 I think it's safest to remove the warning. If we want to add it back later, we can do so for all float types.

**Comment by eric-wieser** [comment]
I don't think we need to fix this for 1.14.1, since 1.13 just gave an error here - so this is already an improvement

**Comment by ahaldane** [comment]
I agree, it's a big improvement since it didn't work before :)

I won't argue too hard since I think the warning will be pretty rare... But still, what is the ultimate desired behavior? I think there is a high chance it is not to raise an overflow warning for any float types. In that case, it would be nice to get the new float128 behavior right, from the get-go in 1.14.

**Comment by eric-wieser** [comment]
Seems like piggy-backing on `np.seterr` could be the right approach here.

I suppose at any rate we should really fix the message in 1.14, even if we leave it in. And if we're touching it...

> First of all, we didn't think about underflow

I missed this in my first read. Are denormal numbers considered underflow? I assumed that word was reserved for non-zero decaying to zero.

**Comment by ahaldane** [comment]
Yes, errno gets set for denormal values (eg, `1e-323 < x < 1e-309` for float64). I just learned that from discussion with the glibc devs.

**Pull request merged**